### PR TITLE
Reorganize Shell Scripts

### LINKED ISSUE #211: Reorganize scripts
### Purpose

Better code organization and readability

### Changes

- Move script to `/scripts/`
   - All sh files except .devcontainer except .devcontainer/deployment
- Use absolute paths instead of relative
- Add README.md in scripts dir that describes what each script does and how to use it
    - Example usages in vscode tasks
    - Add README to docs site

### Resources
- VS Code tasks are defined in code-workspace file
- GitHub CI is defined in `.github/`

### PR DESCRIPTION
**Pull request by chrischang5**
### Description

- Resolves #211 

### Verification

- The CI passes
- Tasks in VSCode work as intended

### CODE REVIEW

File: .devcontainer/deployment/README.md
Comment (patrick-5546) [inline .devcontainer/deployment/README.md:1]: I think we can move `.devcontainer/deployment` to `.scripts/deployment`, what do you think @hhenry01

File: scripts/README.md
Comment (patrick-5546) [inline scripts/README.md:1]: Can briefly descript each script in this directory, like the deployment README does

File: src/network_systems/scripts/run_virtual_iridium.sh
Comment (patrick-5546) [inline src/network_systems/scripts/run_virtual_iridium.sh:1]: What do you think about this @hhenry01: https://github.com/UBCSailbot/sailbot_workspace/issues/211#issuecomment-2076316540

File: .github/actions/run-in-container/ament-lint.sh
Comment (patrick-5546) [inline .github/actions/run-in-container/ament-lint.sh:1]: Move this to scripts dir

File: .github/actions/run-in-container/clang-tidy.sh
Comment (patrick-5546) [inline .github/actions/run-in-container/clang-tidy.sh:1]: Move this to scripts dir

File: .github/actions/run-in-container/test.sh
Comment (patrick-5546) [inline .github/actions/run-in-container/test.sh:1]: Rename to run-tests.sh and move to scripts dir

File: src/network_systems/scripts/run_virtual_iridium.sh
Comment (hhenry01) [reply src/network_systems/scripts/run_virtual_iridium.sh:1]: Yeah that works.

File: .devcontainer/deployment/README.md
Comment (hhenry01) [reply .devcontainer/deployment/README.md:1]: That's fine.

File: scripts/README.md
Comment (chrischang5) [reply scripts/README.md:1]: on it

**Review by patrick-5546 — APPROVED** [review]
lgtm, thanks a lot!

@SPDonaghy fyi, moved all common/ci/deployment scripts to `scripts/` so that everything is one place

**Comment by SPDonaghy** [comment]
Can I merge this one in?

**Comment by chrischang5** [comment]
Yep! Downside of not having consistent internet while on vacation lol

**Pull request merged**